trace-sdk/src/test/java/io/bitrise/trace/data/management/formatter/cpu/SystemCpuDataFormatterTest.java

```diff
@@ -0,0 +1,111 @@
+import static io.bitrise.trace.data.dto.DataValues.steal;
+import static io.bitrise.trace.data.dto.DataValues.system;
+import static io.bitrise.trace.data.dto.DataValues.user;
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Unit tests for {@link SystemCpuDataFormatter}.
+ */
+public class SystemCpuDataFormatterTest extends BaseDataFormatterTest {
+
+    final SystemCpuDataFormatter formatter = new SystemCpuDataFormatter();
+    final CpuUsageData.CpuStat cpuStat = new CpuUsageData.CpuStat(1,2,3,4,5,6,7,8);
+    final String metricDescriptionDescription = "System CPU Usage";
+
+    @Test
+    public void formatData_notCpuStat() {
+        final Data data = new Data(SystemCpuUsageDataCollector.class);
+        data.setContent("cpu usage");
+        assertArrayEquals(new FormattedData[]{}, formatter.formatData(data));
+    }
+
+    @Test
+    public void formatData() {
+        final Data data = new Data(SystemCpuUsageDataCollector.class);
+        data.setContent(cpuStat);
+
+        final FormattedData formattedData = formatter.formatData(data)[0];
+
+        final Timestamp timestamp = formattedData.getMetricEntity().getMetric()
+                .getTimeseries(0).getPoints(0).getTimestamp();
+
+        List<TimeSeries> expectedTimeSeries = new ArrayList<>(Arrays.asList(
+                createCpuTimeSeriesEntry(timestamp, user, cpuStat.getUser()),
+                createCpuTimeSeriesEntry(timestamp, system, cpuStat.getSystem()),
+                createCpuTimeSeriesEntry(timestamp, nice, cpuStat.getNice()),
+                createCpuTimeSeriesEntry(timestamp, idle, cpuStat.getIdle()),
+                createCpuTimeSeriesEntry(timestamp, ioWait, cpuStat.getIoWait()),
+                createCpuTimeSeriesEntry(timestamp, irq, cpuStat.getIrq()),
+                createCpuTimeSeriesEntry(timestamp, softIrq, cpuStat.getSoftIrq()),
+                createCpuTimeSeriesEntry(timestamp, steal, cpuStat.getSteal())));
+
+        final Metric.Builder metricBuilder = Metric.newBuilder();
+        final MetricDescriptor.Builder cpuDescriptorBuilder =
+                MetricDescriptor.newBuilder()
+                        .setDescription(metricDescriptionDescription)
+                        .setName(getName(system, cpu, pct))
+                        .setUnit(percent)
+                        .setType(MetricDescriptor.Type.GAUGE_DOUBLE)
+                        .addLabelKeys(LabelKey.newBuilder()
+                                .setKey(getName(cpu, state))
+                                .build());
+
+        metricBuilder.setMetricDescriptor(cpuDescriptorBuilder.build());
+        metricBuilder.addAllTimeseries(expectedTimeSeries);
+        final Metric expectedMetric = metricBuilder.build();
+
+        assertEquals(expectedMetric, formattedData.getMetricEntity().getMetric() );
+    }
+
+    @NonNull
+    private static TimeSeries createCpuTimeSeriesEntry(@NonNull final Timestamp timestamp,
```

Comment:
We do not have to copy this here, we just have to update the original method in SystemCpuDataFormatter to package default visibility (and add VisibleForTesting annotation). This way we could avoid code duplication.

Note: related to this [comment](https://github.com/bitrise-io/trace-android-sdk/pull/189/files#r622139103), I think the original method should be updated to use double as a param.